src/Compilers/Core/Portable/Generated/Operations.new.Generated.cs

```diff
@@ -0,0 +1,272 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+// < auto-generated />
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.FlowAnalysis;
+
+namespace Microsoft.CodeAnalysis.Operations
```

Comment:
It would be preferable to update the generated file in place, rather than move it chunk by chunk. That would allow reviewers to see the diff.

Reply:
I don't think that will be easier for review, personally.

Reply:
For starters, it would make it clearly visible whether the OperationVisitor changed at all. As the PR stands, there is no easy way to tell.

---
In reply to: [501307338](https://github.com/dotnet/roslyn/pull/48226#discussion_r501307338) [](ancestors = 501307338)

Reply:
OperationVisitor won't be changing. It's a copy-paste currently, and it won't be modified until the very end when I nullable enable everything else.

As to other types, the reason I don't think it will be easier to review is that, particularly when we get to the hierarchies of Base/NonLazy/Lazy types, the diff will end up being far too cluttered to read, as git tries to relate new lines to old lines. They're not really related, and for the most part are uninteresting. The interesting bits are in the files that are not generated.

Reply:
If the removal of some types (Lazy or Base) cause diff to get confused, then the PR should be split into two commits (one that just removes types, one that makes the updates to remaining types).
Starting a new file guarantees that we can't diff, which is definitely worse.

> The interesting bits are in the files that are not generated.

Do you have an example of the change you plan to for the remaining generated code? That might help inform the discussion.

---
In reply to: [501311258](https://github.com/dotnet/roslyn/pull/48226#discussion_r501311258) [](ancestors = 501311258)

Reply:
> If the removal of some types (Lazy or Base) cause diff to get confused, then the PR should be split into two commits (one that just removes types, one that makes the updates to remaining types).

This is not the approach I'm taking. The approach I'm taking is that every commit will change 1 type. This makes the diff very reviewable. I just don't think the diff in this file is particularly useful, and will necessitate adding a bunch of logic to the operation writer around nullable enabling and then disabling for some contexts.

> Do you have an example of the change you plan to for the remaining generated code?

Sure, here's one example: https://github.com/dotnet/roslyn/commit/845e602db46b999e3f1eb18b912c1b45636367bb. Every commit will look something like this.

Reply:
Thanks for the example commit. The diff in that example would be small and easy to review with update-in-place. 
In that example, the interface type is unchanged except maybe for `#nullable` around it and two properties added to actual type. The diff algorithm will keep up fine (such changes are what most of our PRs do). 

If you want to try a diff on a more complex change (with removal of Lazy type), that's an easy experiment. 
I even think we need to include at least a type or two in this first PR to exercise all the code you're adding.

The reason I'm insisting is because I know that many more PRs/commits/types are following. A small investment on the first round will make many reviews easier.

Reply:
>  I even think we need to include at least a type or two in this first PR to exercise all the code you're adding.

I'm not going to put type ports in this PR. Yes, there are a couple of bugs in the generator as written, but they're already fixed and were fixed inline with the commit that discovered them.

Reply:
Since when do we merge code that is not exercised? Is there a reason this PR is special?

Reply:
> Since when do we merge code that is not exercised? Is there a reason this PR is special?

There's nothing new in this PR, no new code. The code that is here is tested, or will be rewritten entirely.

Reply:
> There's nothing new in this PR, no new code.

You're losing me entirely. The old generator has added logic for `PortedTypes` and the new generated code produced different output in the example you shared (except that it's not exercised in this PR, which is a problem).
This discussion is all the more reason to just update the generator and generated files in place. Then it will be clear what is changing. 
I'm expecting the generator to add logic for `if (typeIsPorted) { ... generate new code ... } else { ... generate old code ... }`. The rest of the generator I assume is unchanged, but could not verify easily because of duplication.

Reply:
> new generated code produced different output in the example you shared (except that it's not exercised in this PR, which is a problem).

It did not. There are no logic changes in this PR beyond copy/paste and including/excluding types. The actual changes to the format are going to be in the next PR. Regardless, I've spent most of the day redoing the generator logic and subsequent commits to do in-place transformation, as opposed to side-by-side transformation. I don't think it's particularly nicer or more reviewable, but it'll be what you want.